plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -333,6 +333,9 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
       Logger.e(TAG, "Can't update source: style is not fully loaded.")
       return
     }
+    if (source == null || layer == null) {
+      initLayerAndSource(style)
+    }
```

Comment:
Guess we need to have a test for that now.

Reply:
It's difficult to test it with a test case before we have pixel compare test, only can be done in the example.

Reply:
Can't we have a unit test here somehow to verify that `initLayerAndSource` was called?

Reply:
Rethink about it, we can verify it by verifying `style.addPersistentLayer`, will add a test case later.

Reply:
I guess this PR is the best place to add the test (test that would have failed without changes that this PR is brining and which should start passing now).